src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -744,7 +748,31 @@ private bool TakeStartLine(SocketInput input)
                 RequestUri = requestUrlPath;
                 QueryString = queryString;
                 HttpVersion = httpVersion;
-                Path = RequestUri;
+
+                MemoryPoolIterator2 pathBaseEnd;
+                bool caseMatches;
+
+                if (!string.IsNullOrEmpty(_pathBase) &&
+                    requestUrlPath.Length >= _pathBase.Length &&
+                    pathBegin.StartsWithIgnoreCase(_pathBase, out pathBaseEnd, out caseMatches) &&
```

Comment:
We would like to know if the case matches too. If so, we don't need to allocate a new string and can just do `PathBase = _pathBase`. Otherwise, we have to create a new correctly cased string.